### DISCUSSION TITLE
Github has a new format for their API tokens, and brew doesn't like it!

### DISCUSSION THREAD

**Discussion by jpartain89** [Homebrew/discussions#1201] [Everyday usage] [answered]
So it seems github has changed up their API token and are blasting those of us that are using the now-old API with repeated emails, saying"

```
Hi @<user>, 

We noticed that, at <date and time> your authentication token (<token info>) accessed the GitHub API, the token you used has an outdated format. 

In order to provide additional security benefits to all our customers, we recently updated the format of our API authentication tokens. Please update your authentication tokens to take advantage of these benefits, by visiting github.com/settings/tokens. 

To understand more about this change, and why it's important, visit [developer.github.com/changes/2021-04-05-behind-githubs-new-authentication-token-formats](developer.github.com/changes/2021-04-05-behind-githubs-new-authentication-token-formats). 

Thanks, 
The GitHub Team 
```

Do we know about this? Aware of it? Making updates?

Any info would be awesome

**Comment by carlocab** [comment, marked as answer]
Have you tried regenerating your GitHub token and using the new one with `brew`?

**Reply by jpartain89** [reply]
I ran ```brew search nomachine``` after regenerating the token, didn't change any setting, just hit the regenerate button, updated the info on my machine and got:

```
Warning: Error searching on GitHub: GitHub API Error: Bad credentials
HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
  https://github.com/settings/tokens
```

**Reply by carlocab** [reply]
Did you change `HOMEBREW_GITHUB_API_TOKEN` to the new, regenerated token? You're most likely getting this error because it's still set to the old (invalidated) token.

I regenerated my token recently and have it set to `HOMEBREW_GITHUB_API_TOKEN`, and have not had any problems with it.

**Reply by jpartain89** [reply]
Wow do I feel dumb....

**Reply by Rylan12** [reply]
You need to update the `HOMEBREW_GITHUB_API_TOKEN` environment variable to contain the newly generated token. Have you done that?

**Reply by Rylan12** [reply]
Oops, for some reason my feed didn't reload so I apologize for the duplicate comments

**Comment by Azhrei** [comment]
> Any info would be awesome

So...  While generating my new token per the first answer, I selected only `read: packages` and left everything else turned off.  It _appears_ to work — as in, `brew update` doesn't error out — but I'm not sure if there should've been other options turned on.  Any guidance here?  Thanks.

**Reply by carlocab** [reply]
If it works fine for you then you don't need to do anything. `brew` will complain when it finds your token has inadequate permissions and should be explicit about which additional permissions it needs.
